### ISSUE TITLE
LoopConditionChecker error is not raised by infinite loop

### ISSUE THREAD

**Issue by csrma** [google/error-prone#5077]
The given code runs indefinitely as the variable `i` is nowhere incremented. However, ErrorProne does not raise any error for this.
```
class LoopConditionChecker {
    public int[] showBug(int[] numbers) {
        for (int i = 0; i < numbers.length; ) { // BUG: LoopConditionChecker - the loop condition is never modified in the loop body
            numbers[i] = numbers[i] * 2;
        }
        return numbers;
    }
    
    public static void main(String[] args){
        LoopConditionChecker lcc = new LoopConditionChecker();
        lcc.showBug(new int[]{1, 2, 3});
    }
}
```
This is the output log:
```
warning: [DefaultPackage] Java classes shouldn't use default package
class LoopConditionChecker {
^
    (see https://errorprone.info/bugpattern/DefaultPackage)
1 warning
```

Expectation: LoopConditionChecker is raised for the infinite loop.
ErrorProne version: `2.36.0`
